src/components/layout/RightSectionWallet.vue

```diff
@@ -57,39 +82,55 @@ import AeAmount from '../AeAmount.vue';
 import Dropdown from '../Dropdown.vue';
 import RightSectionTitle from './RightSectionTitle.vue';
 import OutlinedButton from '../OutlinedButton.vue';
+import TokenAvatarAndSymbol from '../fungibleTokens/TokenAvatarAndSymbol.vue';
 
 export default {
   components: {
-    RightSectionTitle, AeAmount, Dropdown, OutlinedButton,
+    RightSectionTitle, AeAmount, Dropdown, OutlinedButton, TokenAvatarAndSymbol,
   },
   props: { closed: Boolean },
   data: () => ({
     walletUrl: process.env.VUE_APP_WALLET_URL,
     hasContractV2Address: !!process.env.VUE_APP_CONTRACT_V2_ADDRESS,
+    showCurrencyDropdown: true,
   }),
   computed: {
     ...mapGetters(['isLoggedIn']),
+    ...mapState('backend', ['prices']),
     ...mapState(['balance', 'address', 'useIframeWallet', 'selectedCurrency', 'tokenBalances']),
-    ...mapState({
-      currencyDropdownOptions({ backend: { prices }, balance }) {
-        return Object.entries(prices).map(([currency, price]) => ({
-          text: [
-            new BigNumber(balance).multipliedBy(price).toFixed(2),
-            currency.toUpperCase(),
-          ].join(' '),
-          value: currency,
-        }));
-      },
-    }),
+    currencyDropdownOptions() {
+      return Object.entries(this.prices).map(([currency, price]) => ({
+        price: new BigNumber(this.balance).multipliedBy(price).toFixed(2),
+        currency,
+      }));
+    },
+    selectedCurrencyPrice() {
+      return new BigNumber(this.balance)
+        .multipliedBy(this.prices[this.selectedCurrency])
+        .toFixed(2);
+    },
+    aeternityTokenData() {
+      return {
+        balance: this.balance,
+        token: '',
+      };
+    },
+    tokenBalancesOptions() {
+      return [
+        this.aeternityTokenData,
+        ...this.tokenBalances,
+      ];
+    },
+  },
+  methods: {
+    ...mapMutations(['updateCurrency', 'enableIframeWallet']),
   },
```

Comment:
```suggestion
  methods: mapMutations(['updateCurrency', 'enableIframeWallet']),
```
Don't change this without need